optuna/storages/base.py

```diff
@@ -29,158 +28,494 @@ class BaseStorage(object, metaclass=abc.ABCMeta):
 
     @abc.abstractmethod
-    def get_study_system_attrs(self, study_id):
-        # type: (int) -> Dict[str, Any]
+    def get_study_user_attrs(self, study_id: int) -> Dict[str, Any]:
+        """Read a dictionary of user-defined attributes of a specified study.
+
+        Args:
+            study_id:
+                Study ID of a study to read.
+        Returns:
+            A dictionary of a user attributes of a study.
+
+        Raises:
+            :exc:`KeyError`:
+                If no study with the matching `study_id` exists.
+        """
+        raise NotImplementedError
 
+    @abc.abstractmethod
+    def get_study_system_attrs(self, study_id: int) -> Dict[str, Any]:
+        """Read a dictionary of optuna-internal attributes of a specified study.
+
+        Args:
+            study_id:
+                Study ID of a study to read.
+        Returns:
+            A dictionary of a system attributes of a study.
+
+        Raises:
+            :exc:`KeyError`:
+                If no study with the matching `study_id` exists.
+        """
         raise NotImplementedError
 
     @abc.abstractmethod
-    def get_all_study_summaries(self):
-        # type: () -> List[study.StudySummary]
+    def get_all_study_summaries(self) -> List[study.StudySummary]:
+        """Read a list of `study.StudySummary` objects.
+
+        Returns:
+            A list of `study.StudySummary` objects.
 
+        """
         raise NotImplementedError
 
     # Basic trial manipulation
 
     @abc.abstractmethod
-    def create_new_trial(self, study_id, template_trial=None):
-        # type: (int, Optional[FrozenTrial]) -> int
-
+    def create_new_trial(
+        self, study_id: int, template_trial: Optional["FrozenTrial"] = None
+    ) -> int:
+        """Create and add a new trial to a specified study.
+
+        Trial ID is unique among all current and deleted trials.
+
+        Args:
+            study_id:
+                Study ID to add a new trial.
+            template_trial:
+                Fronzen trial with default user-attributes, system-attributes,
+                intermediate-values, and a state.
+
+        Returns:
+            Trial ID of the created trial.
+
+        Raises:
+            :exc:`KeyError`:
+                If no study with the matching `study_id` exists.
+        """
         raise NotImplementedError
 
     @abc.abstractmethod
-    def set_trial_state(self, trial_id, state):
-        # type: (int, TrialState) -> bool
-
+    def set_trial_state(self, trial_id: int, state: TrialState) -> bool:
+        """Update a state of a specified trial.
+
+        This method succeeds only when trial is not already finished.
+
+        Args:
+            trial_id:
+                Trial ID of a trial to update its state.
+            state:
+                New state of the trial.
+
+        Returns:
+            :py:obj:`True` if the state is successfully updated.
+            :py:obj:`False` if the state is kept the same.
```

Comment:
A detail but I don't think you need the `:py` part?

Reply:
Removed.